internal/providers/pluginfw/resources/clusters/resource_library.go

```diff
@@ -0,0 +1,237 @@
+package clusters
```

Comment:
Why not put this in `package library`?

Also, we should probably use singular nouns for the package name (e.g. library, not libraries)